eclair-core/src/test/scala/fr/acinq/eclair/router/RouterSpec.scala

```diff
@@ -514,9 +514,72 @@ class RouterSpec extends BaseRouterSpec {
 
     val response = sender.expectMsgType[RouteResponse]
     // the route hasn't changed (nodes are the same)
-    assert(response.routes.head.hops.map(_.nodeId).toList == Seq(a, b, c))
-    assert(response.routes.head.hops.last.nextNodeId == preComputedRoute.targetNodeId)
-    assert(response.routes.head.hops.map(_.lastUpdate).toList == List(update_ab, update_bc, update_cd))
+    assert(response.routes.head.hops.map(_.nodeId) === Seq(a, b, c))
+    assert(response.routes.head.hops.map(_.nextNodeId) === Seq(b, c, d))
+    assert(response.routes.head.hops.map(_.lastUpdate) === Seq(update_ab, update_bc, update_cd))
+  }
+
+  test("given a pre-defined private channels route add the proper channel updates") { fixture =>
+    import fixture._
+    val sender = TestProbe()
+
+    {
+      val preComputedRoute = PredefinedChannelRoute(g, Seq(channelId_ag))
+      sender.send(router, FinalizeRoute(10000 msat, preComputedRoute))
+      val response = sender.expectMsgType[RouteResponse]
+      assert(response.routes.length === 1)
+      val route = response.routes.head
+      assert(route.hops.map(_.lastUpdate) === Seq(update_ag))
+      assert(route.hops.head.nodeId === a)
+      assert(route.hops.head.nextNodeId === g)
+    }
+    {
+      val preComputedRoute = PredefinedChannelRoute(h, Seq(channelId_ag, channelId_gh))
```

Comment:
Are `channelId_ag` and `channelId_gh` private channels? There is no way to tell without digging into other files, maybe renaming `*_ag` to `*_ag_private` would make i more obvious.

Reply:
I agree, I had to go to inside the `BaseRouterSpec` to find a private channel, I'll rename that.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2082/commits/d930bf1777e845773447e1664f0059cc32f18bfc